lib/storage/metadata/mongoclient/MongoClientInterface.ts

```diff
@@ -1143,8 +1143,9 @@ class MongoClientInterface {
             const ops: AnyBulkWriteOperation<ObjectMetastoreDocument>[] = [];
             // filter to get master
             const filter = {
-                '_id': masterKey,
-                'value.versionId': {
+                _id: masterKey,
+                $or: [
```

Comment:
not sure it works, but I was wondering if we could reduce duplication (and maybe make it a bit clearer?) by writing it like this:


```suggestion
                'value.versionId': $or: [
                    { $exists: false },
                    { $gte: mstObjVal!.versionId },
                ]
```

Reply:
No, it's not working. Mongo does not know how to work with `$or` in nested field.

```
db.getCollection("test-or").find({
    'value.versionId': {
        $or: [
            { $exists: false },
            { $gte: 1 },
        ]
    }
})
```

But this one is working:

```
db.getCollection("test-or").find({
    $or: [
        {
            'value.versionId': { $gte: 1 }
        },
        {
            'value.versionId': { $exists: false }
        }
    ]
})
```

Reply:
maybe you can use $eq for the first case?

something like `{ 'value.versionId': { $eq: { $or: [...] }}`

Reply:
@williamlardier not sure to get your point. Something like that ?

```
db.getCollection("test-or").find({
    'value.versionId': {
        $eq: {
            $or: [
                { $exists: false },
                { $gte: 1 },
            ]
        }
    }
})
```

Reply:
yes, but i checked and no the eq do not have any "boolean equality", the $or you have is the best approach I think.
There is the $expr that works, i confirmed it, but this would create aggregations and badly affect perf.